gcloud-java-compute/src/main/java/com/google/cloud/compute/Compute.java

```diff
@@ -0,0 +1,2709 @@
+
+    private final String selector;
+
+    InstanceField(String selector) {
+      this.selector = selector;
+    }
+
+    @Override
+    public String selector() {
+      return selector;
+    }
+  }
+
+  /**
+   * Base class for list filters.
+   */
+  abstract class ListFilter implements Serializable {
+
+    private static final long serialVersionUID = -238638392811165127L;
+
+    private final String field;
+    private final ComparisonOperator operator;
+    private final Object value;
+
+    enum ComparisonOperator {
+      /**
+       * Defines an equals filter.
+       */
+      EQ,
+
+      /**
+       * Defines a not-equals filter.
+       */
+      NE
+    }
+
+    ListFilter(String field, ComparisonOperator operator, Object value) {
+      this.field = field;
+      this.operator = operator;
+      this.value = value;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(field, operator, value);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof ListFilter && toPb().equals(((ListFilter) obj).toPb());
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("field", field)
+          .add("operator", operator)
+          .add("value", value)
+          .toString();
+    }
+
+    String toPb() {
+      return field + ' ' + operator.name().toLowerCase() + ' ' + value.toString();
+    }
+  }
+
+  /**
+   * Class for filtering disk type lists.
+   */
+  class DiskTypeFilter extends ListFilter {
+
+    private static final long serialVersionUID = 4847837203592234453L;
+
+    private DiskTypeFilter(DiskTypeField field, ComparisonOperator operator, Object value) {
+      super(field.selector(), operator, value);
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static DiskTypeFilter equals(DiskTypeField field, String value) {
+      return new DiskTypeFilter(checkNotNull(field), ComparisonOperator.EQ, checkNotNull(value));
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static DiskTypeFilter notEquals(DiskTypeField field, String value) {
+      return new DiskTypeFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns an equals filter for the given field and long value.
+     */
+    public static DiskTypeFilter equals(DiskTypeField field, long value) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2740139423/issues/source?bid=3123104&fileBranchId=3354337#l578)